src/main/java/edu/oregonstate/cs361/battleship/strategy/CleverStrategy.java

```diff
@@ -0,0 +1,43 @@
+package edu.oregonstate.cs361.battleship.strategy;
+
+import java.util.ArrayList;
+
+import edu.oregonstate.cs361.battleship.Coordinate;
+
+public class CleverStrategy implements Strategy {
+
+	/**
+	 * We don't know if we hit last time.
+	 * 
+	 * [x] simple strategy: choose from hit coordinate
+	 * 
+	 * [] more clever: choose from hit coordinate and if we have two or more hits coordinate connected, 
+	 * choose that direction
+	 */
+	@Override
+	public Coordinate shootAtPlayer(ArrayList<Coordinate> playerMisses, ArrayList<Coordinate> playerHits) {
+		
+		for (int i = 0; i < playerHits.size(); i++) {
+			Coordinate coor = playerHits.get(i);
+			
+			// four directions
```

Comment:
In clever strategy, if the firing has hit one of the ships, computer will pick the 4 directions nearby, which will increase the possibility to sink the ship.